docker image fails building for arm64

Was raised on the osm dev list (and is more a hunch as of now, but makes sense I think):

It might be not be possible to build our image on e.g. Apple Silicon. The error is caused by the name of the python bindings when we try to `strip` it:

https://github.com/valhalla/valhalla/blob/3fac9705794f902dceb9b031d04c2f4fa9ada109/Dockerfile#L45

I'd guess it builds actually `python_valhalla.cpython-310-darwin.so` or so.

> I'd guess it builds actually python_valhalla.cpython-310-darwin.so or so.

as I see : 
* `valhalla/python_valhalla.cpython-310-aarch64-linux-gnu.so`
* `valhalla/python_valhalla.cpython-310-x86_64-linux-gnu.so`

-----------------

side note:  

in my (intel) machine the "qemu + Buildx" has been crashed : 
command:  `docker buildx build --pull --progress=plain --platform linux/arm64 -t valhalla-arm64 .  ` 

error:
```
....
#16 10269.6 make[2]: *** [test/CMakeFiles/utrecht_tiles.dir/build.make:77: test/data/utrecht_tiles/traffic.tar] Segmentation fault (core dumped)
#16 10269.6 make[1]: *** [CMakeFiles/Makefile2:6126: test/CMakeFiles/utrecht_tiles.dir/all] Error 2
#16 10269.6 make[1]: *** Waiting for unfinished jobs....
#16 10375.5 [ 93%] Linking CXX shared module valhalla/python_valhalla.cpython-310-aarch64-linux-gnu.so
#16 10390.9 [ 93%] Built target python_valhalla
#16 10831.5 [ 96%] Linking CXX static library libvalhalla_test.a
#16 10835.1 [ 96%] Built target valhalla_test
#16 10835.2 make: *** [Makefile:156: all] Error 2

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 * **Removed**
    * REMOVED: Docker image pushes to Dockerhub [#4033](https://github.com/valhalla/valhalla/pull/4033)
 * **Bug Fix**
+   * FIXED: docker image fails building for arm64 [#3950](https://github.com/valhalla/valhalla/issues/3950)
    * FIXED: underflow of uint64_t cast for matrix time results [#3906](https://github.com/valhalla/valhalla/pull/3906)
    * FIXED: update vcpkg commit for Azure pipelines to fix libtool mirrors [#3915](https://github.com/valhalla/valhalla/pull/3915)
    * FIXED: fix CHANGELOG release year (2022->2023) [#3927](https://github.com/valhalla/valhalla/pull/3927)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,7 +2,6 @@
 * **Removed**
    * REMOVED: Docker image pushes to Dockerhub [#4033](https://github.com/valhalla/valhalla/pull/4033)
 * **Bug Fix**
-   * FIXED: docker image fails building for arm64 [#3950](https://github.com/valhalla/valhalla/issues/3950)
    * FIXED: underflow of uint64_t cast for matrix time results [#3906](https://github.com/valhalla/valhalla/pull/3906)
    * FIXED: update vcpkg commit for Azure pipelines to fix libtool mirrors [#3915](https://github.com/valhalla/valhalla/pull/3915)
    * FIXED: fix CHANGELOG release year (2022->2023) [#3927](https://github.com/valhalla/valhalla/pull/3927)
@@ -19,6 +18,8 @@
    * FIXED: fix build with LOGGING_LEVEL=ALL [#3992](https://github.com/valhalla/valhalla/pull/3992)
    * FIXED: transit stitching when determining whether a platform was generated [#4020](https://github.com/valhalla/valhalla/pull/4020)
    * FIXED: multimodal isochrones [#4030](https://github.com/valhalla/valhalla/pull/4030)
+   * FIXED: Remove Arch-Type from strip command [#4040](https://github.com/valhalla/valhalla/pull/4040)
+
 * **Enhancement**
    * CHANGED: replace boost::optional with C++17's std::optional where possible [#3890](https://github.com/valhalla/valhalla/pull/3890)
    * ADDED: parse `lit` tag on ways and add it to graph [#3893](https://github.com/valhalla/valhalla/pull/3893)

--- a/Dockerfile
+++ b/Dockerfile
@@ -42,7 +42,7 @@ RUN tar -cvf valhalla.debug.tar valhalla_*.debug && gzip -9 valhalla.debug.tar
 RUN rm -f valhalla_*.debug
 RUN strip --strip-debug --strip-unneeded valhalla_* || true
 RUN strip /usr/local/lib/libvalhalla.a
-RUN strip /usr/lib/python3/dist-packages/valhalla/python_valhalla.cpython-310-x86_64-linux-gnu.so
+RUN strip /usr/lib/python3/dist-packages/valhalla/python_valhalla.cpython-310-*-gnu.so
 
 ####################################################################
 # copy the important stuff from the build stage to the runner image

--- a/Dockerfile
+++ b/Dockerfile
@@ -42,7 +42,7 @@ RUN tar -cvf valhalla.debug.tar valhalla_*.debug && gzip -9 valhalla.debug.tar
 RUN rm -f valhalla_*.debug
 RUN strip --strip-debug --strip-unneeded valhalla_* || true
 RUN strip /usr/local/lib/libvalhalla.a
-RUN strip /usr/lib/python3/dist-packages/valhalla/python_valhalla.cpython-310-*-gnu.so
+RUN strip /usr/lib/python3/dist-packages/valhalla/python_valhalla*.so
 
 ####################################################################
 # copy the important stuff from the build stage to the runner image

